### PR TITLE
Discord Authentication and basic functionality

### PR DESCRIPTION
**Pull request by dbgrigsby**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -101,3 +101,8 @@ postr_config.ini
 
 # Config for passwords
 config.py
+
+# Vim swapfiles
+*~
+*.swp
```
Comment (AdamKBeck) [inline .gitignore]: Good idea, swap files were getting annoying to untrack.

File: postr/apis/discord.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        async for msg in client.logs_from(message.channel, limit=1000):
+            if msg:
+                messages_in_channel += 1
+        await client.edit_message(tmp, f'In total, {messages_in_channel} messages in this channel')
+    elif message.content.startswith('!rest'):
+        await client.send_message(message.channel, 'Resting for 5 seconds')
+        await asyncio.sleep(5)
+        await client.send_message(message.channel, 'Done resting')
+    else:
+        await client.add_reaction(message, '👍')
```
Comment (AdamKBeck) [inline postr/apis/discord.py]: Will this emoji break any windows users when looking at the code?

Comment (dbgrigsby) [reply postr/apis/discord.py]: 🤔

Comment (dbgrigsby) [reply postr/apis/discord.py]: I think it'll just display as a box, which would be confusing but not a problem.

In next week's discord finalizing, I'll change all emojis to `\U+ABC123` format, just as a safeguard.

**Pull request merged**